File: src/main/webapp/WEB-INF/static/core/js/service/AplicacaoPerfilService.js
/**
 * AplicacaoPerfilService
 */
app.factory('AplicacaoPerfilService', ['$http', function($http) {
	
	var _getListaPorAplicacao = function(aplicacao) {
		return $http.get(SERVER_APP + '/aplicacao_perfil/get/list/por/aplicacao/'+aplicacao.codigo);
	};
	
	var _getLista = function(aplicacao, perfil) {
		return $http.get(SERVER_APP + '/aplicacao_perfil/get/list/aplicacao/'+aplicacao.codigo+'/perfil/'+perfil.codigo);
	};
	
	var _getPorAppPerfilUsu = function(aplicacao, perfil, usuario) {
		return $http.get(SERVER_APP + '/aplicacao_perfil/get/aplicacao/'+aplicacao+'/perfil/'+perfil+'/usuario/'+usuario);
	};
	
	var _update = function(perfil, usuario, aplicacaoPerfil) {
		return $http.post(SERVER_APP + '/aplicacao_perfil/update/perfil/'+perfil.codigo+'/aplicacao/'+$.trim(perfil.aplicacao.codigo)+'/usuario/'+usuario, aplicacaoPerfil);
	};
	
	var _save = function(aplicacaoPerfil) {
		return $http.post(SERVER_APP + '/aplicacao_perfil/save', aplicacaoPerfil);
	};
	
	return {
		
		getListaPorAplicacao : _getListaPorAplicacao,
		
		getLista : _getLista,
		
		getPorAppPerfilUsu : _getPorAppPerfilUsu,
		
		update : _update,
		
		save : _save
			
	};
	
}]);

File: src/main/java/br/com/tdv/service/AplicacaoPerfilService.java
package br.com.tdv.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.tdv.dao.AplicacaoPerfilDao;
import br.com.tdv.model.Aplicacao;
import br.com.tdv.model.AplicacaoPerfil;
import br.com.tdv.model.Perfil;

@Service("aplicacaoPerfilService")
public class AplicacaoPerfilService {

	@Autowired
	private AplicacaoPerfilDao aplicacaoPerfilDao;

	public List<AplicacaoPerfil> getPorAplicacao(Aplicacao aplicacao) {
		return this.aplicacaoPerfilDao.getPorAplicacao(aplicacao);
	}

	public List<AplicacaoPerfil> getLista(Aplicacao aplicacao, Perfil perfil) {
		return this.aplicacaoPerfilDao.getLista(aplicacao, perfil);
	}

	public AplicacaoPerfil getPorAplicacaoPerfilUsuario(String aplicacaoCodigo, String perfilCodigo, String usuario) {
		return this.aplicacaoPerfilDao.getPorAplicacaoPerfilUsuario(aplicacaoCodigo, perfilCodigo, usuario);
	}

	public void update(AplicacaoPerfil aplicacaoPerfil, String perfil, String aplicacao, String usuario) {
		this.aplicacaoPerfilDao.update(aplicacaoPerfil, perfil, aplicacao, usuario);
	}

	public void save(AplicacaoPerfil aplicacaoPerfil) {
		this.aplicacaoPerfilDao.save(aplicacaoPerfil);
	}

	
	
}
